### ISSUE TITLE
Improve observability of objects and classes

### ISSUE THREAD

**Issue by tflori** [tflori/orm#59]
Currently we are only able from within the classes to define what should happen when an entity gets updated etc.

You could implement a standard event framework and add it to each entity (maybe use a trait) but a simple observer pattern could also be implemented in `EntityManager`. 

I imagine something like:

```php
$em = new EntityManager();
$em->observe(User::class)->on('insert', function (User $user) {});
$em->observe(User::class, new class implements SubscriberInterface {});
```

Another thing could be to provide an existing entity object to `$em->observe($user)` to react on that specific object instead of anyobject of that class.

**Comment by tflori** [comment]
someone might have a look at how observers work now and if he sees any problems:
https://github.com/tflori/orm/commit/bde615c481a9e2219c19a020c2ee68c9076fcd4c#diff-666d121aba2b6880b6f7f886d63eaecb

**Comment by tflori** [comment]
https://github.com/tflori/orm/pull/62 the final version with documentation.. I will give it a try in another project and if no-one complains I will merge and release a new version for that

**Comment by tflori** [comment]
the observers seem to work fine and will be released with version 1.9
